anomalib/deploy/inference.py

```diff
@@ -113,15 +173,57 @@ class TorchInferencer(Inferencer):
         config (DictConfig): Configurable parameters that are used
             during the training stage.
         path (Union[str, Path]): Path to the model ckpt file.
+        meta_data_path (Union[str, Path], optional): Path to metadata file. If none, it tries to load the params
+                from the model state_dict. Defaults to None.
     """
 
-    def __init__(self, config: Union[DictConfig, ListConfig], path: Union[str, Path, AnomalyModule]):
+    def __init__(
+        self,
+        config: Union[DictConfig, ListConfig],
+        path: Union[str, Path, AnomalyModule],
+        meta_data_path: Union[str, Path] = None,
+    ):
         self.config = config
         if isinstance(path, AnomalyModule):
             self.model = path
         else:
             self.model = self.load_model(path)
 
+        self.meta_data = self._load_meta_data(meta_data_path)
+
+    def _load_meta_data(self, path: Optional[Union[str, Path]] = None) -> Union[Dict, DictConfig]:
+        """Load metadata from file or from model state dict.
+
+        Args:
+            path (Optional[Union[str, Path]], optional): Path to metadata file. If none, it tries to load the params
+                from the model state_dict. Defaults to None.
+
+        Returns:
+            Dict: Dictionary containing the meta_data.
+        """
+        meta_data = {}
+        model_params = self.model.state_dict()
+        meta_data_params = {
```

Comment:
Since the model is already loaded here, you could just get the metadata from the model itself:
```
image_threshold = self.model.image_threshold
pixel_threshold = self.model.pixel_threshold
image_mean = self.model.training_distribution.image_mean
...
```